diff -ur bubblewrap.old/bubblewrap.c bubblewrap/bubblewrap.c
--- bubblewrap.old/bubblewrap.c	2025-01-07 14:26:55.401937911 +0100
+++ bubblewrap/bubblewrap.c	2025-01-07 14:32:07.298074128 +0100
@@ -132,6 +132,7 @@
   SETUP_MOUNT_DEV,
   SETUP_MOUNT_TMPFS,
   SETUP_MOUNT_MQUEUE,
+  SETUP_MOUNT_SYS,
   SETUP_MAKE_DIR,
   SETUP_MAKE_FILE,
   SETUP_MAKE_BIND_FILE,
@@ -175,6 +176,7 @@
   PRIV_SEP_OP_BIND_MOUNT,
   PRIV_SEP_OP_OVERLAY_MOUNT,
   PRIV_SEP_OP_PROC_MOUNT,
+  PRIV_SEP_OP_SYS_MOUNT,
   PRIV_SEP_OP_TMPFS_MOUNT,
   PRIV_SEP_OP_DEVPTS_MOUNT,
   PRIV_SEP_OP_MQUEUE_MOUNT,
@@ -351,6 +353,7 @@
            "    --exec-label LABEL           Exec label for the sandbox\n"
            "    --file-label LABEL           File label for temporary sandbox content\n"
            "    --proc DEST                  Mount new procfs on DEST\n"
+           "    --sys DEST                   Mount new sysfs on DEST\n"
            "    --dev DEST                   Mount new dev on DEST\n"
            "    --tmpfs DEST                 Mount new tmpfs on DEST\n"
            "    --mqueue DEST                Mount new mqueue on DEST\n"
@@ -1122,6 +1125,11 @@
         die_with_mount_error ("Can't mount proc on %s", arg1);
       break;
 
+    case PRIV_SEP_OP_SYS_MOUNT:
+      if (mount ("sysfs", arg1, "sysfs", MS_NOSUID | MS_NOEXEC | MS_NODEV, NULL) != 0)
+        die_with_mount_error ("Can't mount sysfs on %s", arg1);
+      break;
+
     case PRIV_SEP_OP_TMPFS_MOUNT:
       {
         cleanup_free char *mode = NULL;
@@ -1367,6 +1375,27 @@
 
           break;
 
+        case SETUP_MOUNT_SYS:
+          if (ensure_dir (dest, 0755) != 0)
+            die_with_error ("Can't mkdir %s", op->dest);
+
+          if (unshare_pid || opt_pidns_fd != -1)
+            {
+              /* Our own procfs */
+              privileged_op (privileged_op_socket,
+                             PRIV_SEP_OP_SYS_MOUNT, 0, 0, 0,
+                             dest, NULL);
+            }
+          else
+            {
+              /* Use system procfs, as we share pid namespace anyway */
+              privileged_op (privileged_op_socket,
+                             PRIV_SEP_OP_BIND_MOUNT, 0, 0, 0,
+                             "oldroot/sys", dest);
+            }
+
+          break;
+
         case SETUP_MOUNT_DEV:
           if (ensure_dir (dest, 0755) != 0)
             die_with_error ("Can't mkdir %s", op->dest);
@@ -1641,6 +1670,7 @@
         case SETUP_RO_OVERLAY_MOUNT:
         case SETUP_TMP_OVERLAY_MOUNT:
         case SETUP_MOUNT_PROC:
+        case SETUP_MOUNT_SYS:
         case SETUP_MOUNT_DEV:
         case SETUP_MOUNT_TMPFS:
         case SETUP_MOUNT_MQUEUE:
@@ -2101,6 +2131,17 @@
           op->dest = argv[1];
 
           argv += 1;
+          argc -= 1;
+        }
+      else if (strcmp (arg, "--sys") == 0)
+        {
+          if (argc < 2)
+            die ("--sys takes an argument");
+
+          op = setup_op_new (SETUP_MOUNT_SYS);
+          op->dest = argv[1];
+
+          argv += 1;
           argc -= 1;
         }
       else if (strcmp (arg, "--exec-label") == 0)
